posts/2021-01-06-ocp-log-forwarding.adoc

```diff
@@ -324,9 +324,19 @@ Proceed to the `Review` section and submit. Copy the generated token value to us
+
+For more information regarding Open Libety logging see the Open Liberty's link:https://openliberty.io/docs/20.0.0.12/log-trace-configuration.html[loging and tracing documentation]. For specifics regarding JSON logging, see the link:https://openliberty.io/docs/20.0.0.12/log-trace-configuration.html#json[Open Liberty JSON logging documentation]. Also see thelink:https://openliberty.io/docs/20.0.0.12/log-trace-configuration.html#json[JSON log events reference list] for a complete index of available fields from the different log types.
+
+Note that JSON logging is configured by default with all log types enabled if you are deploying Open Liberty through the Open Liberty Operator. See link:https://github.com/OpenLiberty/open-liberty-operator/blob/d4f3f2dddad8c97f9e0deb74e1051d4c2bc27694/doc/user-guide.adoc[the Open Liberty Operator documentation] for more information on how to configure the operator to suit your needs.
+
+=== Configuring OpenShift to parse Open Liberty JSON logs
+
+If you do not intend to use Open Liberty JSON logging to leverage the use of dashboards then you may proceed to <<viewing-logs-in-splunk,Viewing logs in Splunk>>.
+
+In OpenShift, the cluster logging Fluentd collectors capture the container logs and sets each log into a message field of a Fluentd JSON document. This JSON is sent to their destination log consumer. To properly leverage Open Liberty JSON log data in a dashboard, you must configure the Fluentd collectors to merge nested JSON data to the parent JSON payload. Although this configuration is required to use the Liberty dashboards, it can cause errors in the default OpenShift Elasticsearch instance. Elasticsearch's dynamic mapping indexes a field and data type on a first-come basis. If multiple deployments generate JSON logs with the same fields, but are of different data types, merging the JSON data can potentially lead to data loss in the default Elasticsearch instance. Review your OpenShift configuration to determine whether the potential impact on the default Elasticsearch instance poses a risk to your environment. The following are the steps required to enable merging of nested JSON fields.
```

Comment:
Although this configuration is required to use the Liberty dashboards, it can cause errors in the default OpenShift Elasticsearch instance. Elasticsearch's dynamic mapping indexes a field and data type on a first-come basis. 
->
Enabling Fluentd to merge JSON logs can cause problems if JSON fields from different products or applications use the same JSON field names to represent different data types.  This is because Elasticsearch uses the first occurrence of a field to determine the data types corresponding to that field name.